app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/MainActivityTest.kt

```diff
@@ -288,20 +311,90 @@ class MainActivityTest {
+            onView(withId(R.id.mainDrawerLayout))
+                .perform(DrawerActions.open(GravityCompat.END))
+
+            onView(withId(R.id.mainNavDrawContact))
+                .perform(scrollTo())
+                .check(matches(isDisplayed()))
+                .perform(click())
+
+            // Enter a message into the message EditText
+            onView(withId(R.id.message_edittext)).perform(typeText("Test message 1"), closeSoftKeyboard())
+
+            // Check if the message EditText has the correct text
+            onView(withId(R.id.message_edittext)).check(matches(withText("Test message 1")))
+
+            onView(withId(R.id.send_button))
+                .check(matches(isDisplayed()))
+                .perform(click())
+
+            // Check that the database function got called with the right arguments
+            verify(db).addFeedback(eq("Test message 1"), eq(userId1), org.mockito.kotlin.any(), org.mockito.kotlin.any())
+        }
+    }
+
+    @Test
+    fun testContactUsBottomSheetMessageStillVisibleWhenWriteFails() {
+        runTest {
+            `when`(db.addFeedback(org.mockito.kotlin.any(), org.mockito.kotlin.any() , org.mockito.kotlin.any(), org.mockito.kotlin.any()))
+                .thenThrow(DatabaseException("error"))
+            // sign in user
+            authUserFlow.emit(userId1)
+
+            // Open the navigation drawer
+            onView(withId(R.id.mainDrawerLayout))
+                .perform(DrawerActions.open(GravityCompat.END))
+
+            onView(withId(R.id.mainNavDrawContact))
+                .perform(scrollTo())
+                .check(matches(isDisplayed()))
+                .perform(click())
+
+            // Enter a message into the message EditText
+            onView(withId(R.id.message_edittext)).perform(typeText("Test message 2"), closeSoftKeyboard())
+
+            // Check if the message EditText has the correct text
+            onView(withId(R.id.message_edittext)).check(matches(withText("Test message 2")))
+
+            // Scroll to the Send button
+            onView(withId(R.id.send_button))
+                .check(matches(isDisplayed()))
+                .perform(click())
 
-            // The test commented below are working locally but not with the CI
-//            // Scroll to the Send button
-//            onView(withId(R.id.send_button))
-//                .check(matches(isDisplayed()))
-//                .perform(click())
-//
-//            // Check that the Bottom Sheet Dialog was dismissed
-//            onView(withId(R.id.bottom_sheet_contact_us)).check(doesNotExist())
+            // Check that the Bottom Sheet Dialog was dismissed
+            onView(withId(R.id.bottom_sheet_contact_us)).check(matches(isDisplayed()))
 
+            // Check that the database function got called with the right arguments
+            verify(db).addFeedback(eq("Test message 2"), eq(userId1), org.mockito.kotlin.any(), org.mockito.kotlin.any())
```

Comment:
This line is not mandatory as it will probably test exactly the same thing as the upper test. What do you think?

Reply:
It is just a double check indeed, don't you think it is okay to keep it?

Reply:
It's just slowing down the CI but I have no problem keeping it